**Лабораторная работа 3**
---

*Доверительные интервалы и корреляционный анализ*


*Задачи должны быть оформлены с выводом промежуточных вычислений (среднее значение, отклонения, расчет критерия.*

---



**1.**

Даны значения величины заработной платы заемщиков банка (zp) и значения их
поведенческого кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

• Найдите ковариацию этих двух величин с помощью элементарных действий, а затем
с помощью функции cov из numpy.

• Полученные значения должны быть равны (если не равны, то обосновать, почему так получилось и как решить данную проблему).

• Найдите коэффициент корреляции Пирсона с помощью ковариации и
среднеквадратичных отклонений двух признаков.


In [ ]:
import numpy as np
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])        #значения величины заработной платы заемщиков банка
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])   #значения их поведенческого кредитного скоринга 
среднее_зп = zp.mean()                                              #среднее значение заработной платы
среднее_кс = ks.mean()                                              #среднее значение кредитного скоринга
среднее_КСЗП = (zp * ks).mean()                                     #среднее от перемножения 2х списков
ковариация_1 = среднее_КСЗП - среднее_зп * среднее_кс
print("Ковариация, вычисленная с помощью элементарных действий = ", ковариация_1)

print("Ковариационная матрица, вычисленная с помощью numpy", np.cov(zp, ks, ddof=0))
ковариация_2 = ((zp - zp.mean()) * (ks - ks.mean())).mean()
print("Ковариация другим методом = ", ковариация_2)
СКО_зп = zp.std()                                                   #среднеквадратическое отклонение заработной платы
print("Среднеквадратическое отклонение заработной платы    = ",СКО_зп)                                               
СКО_кс = ks.std()                                                   #среднеквадратическое отклонение кредитного скоринга
print("Среднеквадратическое отклонение кредитного скоринга = ", СКО_кс)
пирсон = ковариация_1 / (СКО_зп * СКО_кс)                           #коэффициент корреляции Пирсона
print("Коэффициент корреляции Пирсона = ", пирсон)
np.corrcoef(zp, ks)                                                 #коэффициент корреляции с помощью функции в numpy

Ковариация, вычисленная с помощью элементарных действий =  9157.839999999997
Ковариационная матрица, вычисленная с помощью numpy [[ 3494.64  9157.84]
 [ 9157.84 30468.89]]
Ковариация другим методом =  9157.84
Среднеквадратическое отклонение заработной платы    =  59.115480206118605
Среднеквадратическое отклонение кредитного скоринга =  174.55340157098058
Коэффициент корреляции Пирсона =  0.8874900920739158


array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

**Ответ:** 
Ковариация 2-х величин составляет 9157.84

Коэффициент корреляции Пирсона =  0.88749009


---



**2.**

Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111. 

Известно, что в генеральной совокупности
IQ распределен нормально.

• Найдите доверительный интервал для математического ожидания с надежностью
0.95.

In [ ]:
from scipy import stats
IQ = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])   #значения IQ выборки студентов
n_IQ = len(IQ)                                                     #количество значений
alpha_IQ = 0.05                                                    #надежность 95%
среднее_IQ = IQ.mean()                                             #cреднее
print("Среднее значение =", среднее_IQ)                                             
СКО_IQ = IQ.std(ddof=1)                                            #среднеквадратическое
print("СКО =", СКО_IQ)
t_IQ = stats.t(n_IQ - 1).ppf(1 - alpha_IQ/2)                       #t-критерий
print("t =",t_IQ)
границы_IQ = (среднее_IQ - t_IQ * СКО_IQ/np.sqrt(n_IQ), среднее_IQ + t_IQ * СКО_IQ/np.sqrt(n_IQ))
print("Границы:", границы_IQ)

Среднее значение = 118.1
СКО = 10.54566788359614
t = 2.2621571627409915
Границы: (110.55608365158724, 125.64391634841274)


**Ответ:** доверительный интервал для математического ожидания с надежностью 0.95 находится в интервале от 110.55608365158724 до 125.64391634841274


---



**3.**

Известно, что рост футболистов в сборной распределен нормально с дисперсией
генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее
выборочное составляет 174.2. 

Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [ ]:
alpha_рост = 0.05                                     #надежность 95%
n_рост = 27                                           #количество значений
среднее_рост = 174.2                                  #среднее значение
дисперсия_рост = 25                                   #дисперсия
СКО_рост = np.sqrt(дисперсия_рост)                    #среднеквадратическое отклонение
t_рост = stats.t(n_рост - 1).ppf(1 - alpha_рост/2)    #t-критерий
print("t =",t_рост)
границы_рост = (среднее_рост - t_рост * СКО_рост/np.sqrt(n_рост), среднее_рост + t_рост * СКО_рост/np.sqrt(n_рост))
print("Границы:", границы_рост)

t = 2.055529438642871
Границы: (172.2220658754539, 176.17793412454608)


**Ответ:** доверительный интервал для математического ожидания с надежностью 0.95 находится в интервале от 172.2220658754539 до 176.17793412454608


---

